### PR TITLE
fix #50 parsing literals with escaped quote

### LINKED ISSUE #50: Parsing Literals with escaped quote fails
Any CQL statement with a string containing an escaped quote

Example: 

```cql
SELECT entry_title, content FROM posts WHERE blog_title='John''s Blog'
```

### PR DESCRIPTION
**Pull request by SalmaKhater**

### CODE REVIEW

File: cql-parser/src/test/scala/troy/cql/parser/dml/SelectStatementParserTest.scala [hunk 1]
```diff
@@ -295,6 +295,35 @@ class SelectStatementParserTest extends FlatSpec with Matchers {
...
+
+    selection.items(1).selector shouldBe Select.ColumnName("content")
+    selection.items(1).as.isEmpty shouldBe true
+
+    statement.where.isDefined shouldBe true
+    val relations = statement.where.get.relations
+    relations.size shouldBe 1
+    relations(0).asInstanceOf[Relation.Simple].columnName shouldBe "blog_title"
+    relations(0).asInstanceOf[Relation.Simple].operator shouldBe Operator.Equals
+    relations(0).asInstanceOf[Relation.Simple].term shouldBe Constant("John''s Blog")
```
Comment (tabdulradi) [inline cql-parser/src/test/scala/troy/cql/parser/dml/SelectStatementParserTest.scala]: I am not quite sure if this should be "John''s Blog" or "John's Blog".

Comment (SalmaKhater) [inline cql-parser/src/test/scala/troy/cql/parser/dml/SelectStatementParserTest.scala]: fixed

File: cql-parser/src/test/scala/troy/cql/parser/dml/SelectStatementParserTest.scala [hunk 2]
```diff
@@ -295,6 +295,35 @@ class SelectStatementParserTest extends FlatSpec with Matchers {
     relations(0).asInstanceOf[Relation.Simple].term shouldBe Constant("50.4")
   }
 
+  it should "parse select statements with simple where clause with escaped quote" in {
+    val statement = parseQuery(
+      "SELECT entry_title, content FROM posts WHERE blog_title='John''s Blog';"
```
Comment (tabdulradi) [inline cql-parser/src/test/scala/troy/cql/parser/dml/SelectStatementParserTest.scala]: Would this also match multiple escaped quotes in the same query? like:
```
"SELECT entry_title, content FROM posts WHERE blog_title='This isn''t John''s Blog';"
```

Comment (tabdulradi) [inline cql-parser/src/test/scala/troy/cql/parser/dml/SelectStatementParserTest.scala]: Let's try
```
"'".r ~> """(^'||'')*""".r <~ "'"
```
then `map` and replace any `''` with `'`

Comment (SalmaKhater) [inline cql-parser/src/test/scala/troy/cql/parser/dml/SelectStatementParserTest.scala]: The idea is correct but ur syntax isn't correct 😄

**Pull request merged**